packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -12,15 +16,21 @@ interface IPoolMetadataRegistry {
 /// @title A Pool Metadata dApp
 /// @author Bleu LLC
 /// @notice This contract provides a registry for Balancer's Pools metadata
-contract PoolMetadataRegistry is IPoolMetadataRegistry {
+contract PoolMetadataRegistry is IPoolMetadataRegistry, PoolMetadataAuthorization {
     IVault private immutable _vault;
-
+    bytes32 private _poolId;
```

Comment:
hmm, it sounds confusing to me declaring this `_poolId` -- IIRC it's required for `_getOwner` function?

Reply:
yes, to get the pool owner I need the poolId. 

1. The authenticate modifier calls the canPerform. 
2. the canPerform just receive the `bytes32 actionId` and the `address account`
3. the canPerform calls getOwner without any arguments

Any idea of how to solve it with another approach?